packages/destination-actions/src/destinations/tiktok-audiences/functions.ts

```diff
@@ -36,14 +44,61 @@ export async function processPayload(
     // Segment just created isn't available yet for updates via this endpoint.
     // Audiences are usually available to accept batches of data 1 - 2 minutes after
     // they're created. Here, we'll throw an error and let Centrifuge handle the retry.
-    if (res.status !== 200) {
+    if (res.status === 400 && !payloads[0].audience_id && res.data.message.includes('please send valid audience ID')) {
       throw new RetryableError('Error while attempting to update TikTok Audience. This batch will be retried.')
     }
   }
 
   return res
 }
 
+// TikTok returns a max of 100 audiences per request to their `list` endpoint.
+// A customer can have a max of 400 audiences in a single advertiser account.
+// We may have to make up to 4 requests to get all audiences.
+// The first request will return the total_number of audiences associated with
+// the advertiser account.
+export async function getAllAudiences(TikTokApiClient: TikTokAudiences) {
+  let response = await TikTokApiClient.getAudiences(1, 100)
+  let audiences: Audiences[] = response.data.list
+  const total_number_audiences = response.data.page_info.total_number
+  let recieved_audiences = response.data.page_info.page_size
+  let page_number = 2
```

Comment:
consider a meaningful name so the code conveys its behavior.
```suggestion
  let page_number = PAGINATION_START_INDEX
```